tests/sentry/usage_accountant/test_accountant.py

```diff
@@ -35,6 +36,7 @@ def assert_msg(
     }
 
 
+@pytest.mark.skip(reason="usage of atexit breaks interpreter")
```

Comment:
got tagged in this via codeowners, i think if you replace `atexit._run_exitfuncs()` in the test with `accountant._shutdown()` you should be good

Reply:
agreed -- but it shouldn't be using atexit at all and should get rewritten to a context manager or some such

Reply:
yeah I mean more that accountant shouldn't use atexit -- if it's actually for COGS data `atexit` is unreliable-at-best

Reply:
@fpacifici can speak more to this, but the intent is to flush only once per lifecycle of a process. i.e. you have multiple celery tasks submitting messages, and then once the process is reaped you want to flush out the batch.

there is no other way to do this in celery that I know of, or at least they are all similar (celery signal or atexit handler) unless you want to flush on every message and leave nothing to flush on process shutdown. that would have an impact on overhead of those measurements.

Reply:
imo if you want to measure the lifetime of a process, wrap the process